pkg/csi/service/vanilla/controller.go

```diff
@@ -679,19 +679,16 @@ func (c *controller) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequ
 				},
 			})
 			if err != nil {
-				msg := fmt.Sprintf("QueryVolume failed for volumeID: %q. %+v", req.VolumeId, err.Error())
-				log.Error(msg)
-				return nil, status.Error(codes.Internal, msg)
-			}
-			if len(queryResult.Volumes) == 0 {
-				msg := fmt.Sprintf("volumeID %s not found in QueryVolume", req.VolumeId)
+				msg := fmt.Sprintf("QueryAllVolume failed for volumeID: %q. %+v", req.VolumeId, err.Error())
 				log.Error(msg)
 				return nil, status.Error(codes.Internal, msg)
```

Comment:
If `QueryAllVolume` call fails, do we want to return error or just ignore it?

Reply:
I thought about it, it can fail if CNS QueryAll API fails. I thought we should throw back the error as that indicates a problem in CNS. But I'm ok to ignore it as well as we are doing a best effort to find the volume type. @chethanv28 @divyenpatel @gohilankit what do you guys think?

Reply:
IMO, we can ignore it

Reply:
I think we can ignore the error here. If there are issues connecting to CNS, the operation will anyway fail and return an error, which we will be capturing in Prometheus as `unknown` volume type either ways.

Reply:
Discussed offline with Divyen. For now we will not invoke QueryAll API. Instead the DeleteVolume metric will be pushed with "unknown" volume type. We will revisit this in the future when we use CR for orphan volumes cleanup.